pkg/api/v1beta1/clustercurator_types.go

```diff
@@ -60,6 +60,28 @@ type Hooks struct {
 	OverrideJob *runtime.RawExtension `json:"overrideJob,omitempty"`
 }
 
+type UpgradeHooks struct {
+
+	// DesiredUpdate indicates the desired value of
+	// the cluster version. Setting this value will trigger an upgrade (if
+	// the current version does not match the desired version).
+	// +optional
+	DesiredUpdate string `json:"desiredUpdate,omitempty"`
+
+	// Channel is an identifier for explicitly requesting that a non-default
+	// set of updates be applied to this cluster. The default channel will be
+	// contain stable updates that are appropriate for production clusters.
+	// +optional
+	Channel string `json:"channel,omitempty"`
```

Comment:
where will hub users get the channel for the job? currently MCI only has available versions

and what is `non-default set of updates`? does that mean the field value should be default/non-default

Reply:
I copied this from clusterversions. I think default means the channel which is already in clusterversion resource